src/main/java/com/microsoft/azure/relay/HybridHttpConnection.java

```diff
@@ -233,10 +234,12 @@ void invokeRequestHandler(RequestCommandAndStream requestAndStream) {
 		}
 	}
 
-	private CompletableFuture<Void> sendBytesOverRendezvousAsync(ByteBuffer buffer, Duration timeout) {
-		int bytesToWrite = buffer.remaining();
-		return this.rendezvousWebSocket.writeAsync(buffer, timeout).thenAccept(nullResult -> {
-			RelayLogger.logEvent("httpSendingBytes", this, String.valueOf(bytesToWrite));
+	private CompletableFuture<Void> sendBytesOverRendezvousAsync(ByteArrayOutputStream writeBufferStream, Duration timeout) {
+		if (writeBufferStream == null) {
+			return CompletableFuture.completedFuture(null);
+		}
+		return this.rendezvousWebSocket.writeAsync(writeBufferStream.toByteArray(), timeout).thenAccept(nullResult -> {
+			RelayLogger.logEvent("httpSendingBytes", this, String.valueOf(writeBufferStream.size()));
```

Comment:
>size [](start = 83, length = 4)

does size() still return the correct length after toByteArray was called?

Reply:
size() returns the number of effective bytes in the underlying byte[], so it will still reference to the same thing and return the same result.

---
In reply to: [276023732](https://github.com/Azure/azure-relay-java/pull/39#discussion_r276023732) [](ancestors = 276023732)